pkg/migrate/cstor/cstorvolumepolicy.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrate
+
+import (
+	"strconv"
+	"strings"
+
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"gopkg.in/yaml.v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type config struct {
+	Name  string `yaml:"name"`
+	Value string `yaml:"value"`
+}
+
+type quantity struct {
+	Memory string `yaml:"memory,omitempty"`
+	CPU    string `yaml:"cpu,omitempty"`
+}
+
+// createCVPforConfig creates an equivalent CVP
+// for the cas config annotation set on old SC
+func (v *VolumeMigrator) createCVPforConfig(sc *storagev1.StorageClass) error {
+	_, err := v.OpenebsClientset.CstorV1().
+		CStorVolumePolicies(v.OpenebsNamespace).
+		Get(sc.Name, metav1.GetOptions{})
+	found := false
+	if err == nil {
+		found = true
+	}
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	scConfig := sc.Annotations["cas.openebs.io/config"]
```

Comment:
We also need to check for [`Target Affinity Policy`](https://docs.openebs.io/docs/next/ugcstor.html#Target-Affinity-Policy). If SC has this annotation then we have to set `policy.Spec.Provision.ReplicaAffinity` to true. It will help when the user scaled up the sts then replica also needs to schedule on the same node.
cc: @prateekpandey14

Reply:
As this CVP will be not be used for migrated volume and only for new provisioning we can take this change.